docs/story_tables.md

```diff
@@ -0,0 +1,187 @@
+3. For each page
+  1. Identify and store all the fields on the page and all the variables they set.
+  2. Shallowly clone the Story Table row list.
+    1. Create a new object almost the same as the old object
+    2. Print a warning to the user if they've created a `.there_is_another` row with a value of `'True'` and change that value to `'False'`.
+    3. Add artificial rows - rows we create - which is currently just used for `.target_number`.
+    4. This new list will be used to actually put answers into fields.
+  3. For each field
+    1. Find which Story Table row, if any, matches that field.
+    2. Set the field to the given value.
+    3. Increment the row's accumulators.
+      1. Increment all relevant `.times_used` by 1.
+      2. Increment the relevant `.used_for['foo']` by 1.
+  4. Try to continue.
+
+Now some details.
+
+A row object can have two different structures (sorry). The first is for non-artificial rows:
+
+```js
+{
+  original: {
+    trigger: `moms.target_number`,
+    var: `x.target_number`,
+    value: `SECRET_NUMBER_OF_MOMS`,
+  },
+  trigger: `moms.target_number`,
+  var: `x.target_number`,
+  // These are different
+  value: `2`,
+  is_artificial: false,
+  times_used: 0,
+  used_for: { `x.target_number`: 0 },
+}
+```
+
+The second is for artificial rows. An example of a `.there_are_any` row:
+
+```js
+{
+  source: {
+    // The contents of the previous outer object, with one change:  
+    used_for: {
+      `x.target_number`: 0,
+      `x.there_are_any`: 0,
+      `x.there_is_another`: 0,
+    }
+  }
+  is_artificial: true,
+  var: `x.there_are_any`,
+  value: `True`,
+  trigger: `moms.there_are_any`,
+  used_for: { `x.there_are_any`: 0 }
+  // `.times_used` is the same
+}
+```
+
+An example of a `.there_is_another` row:
+
+```js
+{
+  source: {
+    // The contents of the previous outer object, with one change:  
+    used_for: {
+      `x.target_number`: 0,
+      `x.there_are_any`: 0,
+      `x.there_is_another`: 0,
+    }
+  }
+  is_artificial: true,
+  var: `x.there_is_another`,
+  value: `True`,
+  trigger: `moms.there_is_another`,
+  used_for: { `x.there_is_another`: 0 }
+  //`.times_used` is the same
+}
+```
+
+Why is it all so nested? That's just how it evolved. It could probably use some refactoring.
+
+The parts relevant to loops for gathering multiple items in a list are the `.target_number` accumulators and the artificial rows. Whenever a `.there_is_another` row is used to set an answer to a field, we add 1 to its `source.times_used`. That is, we mutate the `.target_number` object's `.times_used` property. We alsoadd 1 to its `source.used_for['x.there_is_another']` value. That mutates the `'x.there_is_another'` property of the `.used_for` of the `.target_number` object. We do the same for `.there_is_another` rows, but that doesn't matter for now.
+
+Why do we increment both the `.times_used` property and the `.used_for` properties? Have patience and read on.
```

Comment:
I do think it's good to address `.times_used` when it first appears. I wanted to just highlight it in people's minds. Also, that's one of the questions I'd have if I were to see that property. Maybe that's a comment that can be added to the code as well. The below is redundant, but maybe it's better than skipping the property altogether. Thoughts?

```suggestion
It would seem like the `.times_used` property is redundant, but it has a different purpose - it lets us know whether the `target_number` row was used at all during the test to let us give appropriate feedback to the user.
```

Reply:
For posterity, I've moved this elsewhere.